src/pl-fli.c

```diff
@@ -1489,7 +1489,18 @@ PL_get_bool(term_t t, int *b)
     { *b = FALSE;
       succeed;
     }
+    fail;
   }
+  if ( isInteger(w) )
```

Comment:
This is problematic.   First of all this should be in PL_get_bool() as PL_get_bool_ex() is merely an exception-raising version of PL_get_bool().  We do not want other differences.   

More dubious however is whether we should allow anything other than 0 or 1?   If not, we are entering the field of what I think is called _generalized Booleans_ (not sure about the term) which for many types defines some member of the type to be `false` and thus the others to be `true`.    So, we get floats (0.0), strings (""), atoms (''), and what more?    I don't think that is the direction we should go.

I don't really see the need for anything but `true` and `false`.  `on and `off` came there because Prolog flags tended to use `on` and `off` in many systems (still do, I think).   That makes `0` and `1` acceptable.

If we we go that way we must also deal with bigints (that are always `true` because integers always use the smallest representation that can represent them when they live on the stacks).

Reply:
This *is* in PL_get_bool(). 

I was of two minds about 0/not-0 vs 0/1 for boolean. I'll change it to 0/1. That will require a small change to PL_get_bool_ex(), to determine whether the cause of failure is due to not being one of the allowed atoms (true,false,on,off) or being a representation error (not atom / {0,1}).